gcs/src/main/scala/io/monix/connect/gcs/Blob.scala

```diff
@@ -0,0 +1,163 @@
+   *
+   * $copyToNote
+   */
+  def copyTo(targetBucket: String, options: BlobSourceOption*): Task[Blob] =
+    Task.evalAsync(underlying.copyTo(targetBucket, options: _*))
+      .map(_.getResult)
+      .map(Blob.apply)
+
+  /**
+   * Copies this blob to the target Blob in the target Bucket.
+   *
+   * $copyToNote
+   */
+  def copyTo(targetBucket: String, targetBlob: String, options: BlobSourceOption*): Task[Blob] =
+    Task.evalAsync(underlying.copyTo(targetBucket, targetBlob, options: _*))
+      .map(_.getResult)
+      .map(Blob.apply)
+
+  /**
+   * Generates a signed URL for this blob. If you want to allow access for a fixed amount of time to
+   * this blob, you can use this method to generate a URL that is only valid within a certain time
+   * period. This is particularly useful if you don't want publicly accessible blobs, but also don't
+   * want to require users to explicitly log in. Signing a URL requires a service account signer.
+   *
+   * If an instance of [[com.google.auth.ServiceAccountSigner]] was passed to [[com.google.cloud.storage.StorageOptions]]
+   * builder via [[com.google.cloud.storage.StorageOptions#setCredentials]] or the default credentials are being
+   * used and the environment variable 'GOOGLE_APPLICATION_CREDENTIALS' is set or your application is running in
+   * App Engine, then this function will use those credentials to sign the URL.
+   *
+   * If the credentials passed to [[com.google.cloud.storage.StorageOptions]] do not implement
+   * [[com.google.auth.ServiceAccountSigner]] (this is the case, for instance, for Compute Engine credentials and
+   * Google Cloud SDK credentials) then this function will throw an [[IllegalStateException]] unless an implementation
+   * of [[com.google.auth.ServiceAccountSigner]] is passed using the [[SignUrlOption#signWith(ServiceAccountSigner)]]
+   * option.
+   */
+  def signUrl(duration: FiniteDuration, options: SignUrlOption*): Task[URL] =
+    Task(underlying.signUrl(duration.toMillis, TimeUnit.MILLISECONDS, options: _*))
+
+  /**
+   * Creates a new ACL entry on this Blob.
+   */
+  def createAcl(acl: Acl): Task[Acl] =
+    Task(underlying.createAcl(acl))
+
+  /**
+   * Returns the [[Acl]] entry for the specified entity on this [[Blob]] or [[None]] if not found.
+   */
+  def getAcl(acl: Acl.Entity): Task[Option[Acl]] =
+    Task(underlying.getAcl(acl)).map(Option(_))
+
+  /**
+   * Updates an ACL entry on this [[Blob]].
+   */
+  def updateAcl(acl: Acl): Task[Acl] =
+    Task(underlying.updateAcl(acl))
+
+  /**
+   * Deletes the [[Acl]] entry for the specified [[com.google.cloud.storage.Acl.Entity]] on this [[Blob]].
+   */
+  def deleteAcl(acl: Acl.Entity): Task[Boolean] =
+    Task(underlying.deleteAcl(acl))
+
+  /**
+   * Returns an [[Observable]] of all the [[Acl]] Entries for this [[Blob]].
+   */
+  def listAcls(): Observable[Acl] = {
+    Observable
+      .fromTask(Task(underlying.listAcls()))
+      .concatMapIterable(_.asScala.toList)
```

Comment:
What about just: `Observable.from(underlying.listAcls())`

Reply:
From what I can see this is making an HTTP call to the Google APIs. So wanted to wrap it in a `Task`

Reply:
`Observable` is also an effect in its own right and can suspend side effects, e.g.
```scala
Observable.suspend {
  Observable.fromIterable(sideEffectingCall())
}
```

Reply:
Awesome :-). I will update :-).